codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -32,6 +33,8 @@ def __init__(self, environment):
         self.timer = environment.get("timer")
         self.metadata = environment["agent_metadata"]
         self.agent_config_merger = environment["agent_config_merger"]
+        self.is_lambda_one_click_pg_created_during_execution = False
+        # self.is_create_pg_called_during_submit_profile = False
```

Comment:
Besides having all these flags for the Lambda usecase, how about creating a PG (if it doesn't exist already) in the Lambda decorator only? https://github.com/aws/amazon-codeguru-profiler-python-agent/blob/main/codeguru_profiler_agent/aws_lambda/profiler_decorator.py

Reply:
> I suppose you want to keep only one of this `is_create_pg_called_during_submit_profile` and remove the other.

Good catch! Forgot to remove this. Fixed in the latest commit

Reply:
> Besides having all these flags for the Lambda usecase, how about creating a PG (if it doesn't exist already) in the Lambda decorator only? https://github.com/aws/amazon-codeguru-profiler-python-agent/blob/main/codeguru_profiler_agent/aws_lambda/profiler_decorator.py

We could use that approach but we do not want the initialization of the decorator to depend on the API calls. If let's say we do that, then we will have to check whether a PG has been created already by using DescribeProfilingGroup API and then conditionally call CreateProfilingGroup API. Then, in this case the load on control plane API describePG will be huge if the lambda scales to lets say 10000 instances. Another variation of this would be to call CreateProfilingGroup API unconditionally, in which case the load on CreatePG will be huge. Instead if we depend on refresh_configuration() or report() method calls, then we automatically get the functionality where if PG is not present already then we get ResourceNotFoundException, and we get the functionality where we will get this exception only once and we will call CreatePG only once.

Reply:
Thank you for the explanation, good point!